gui/wxpython/vdigit/toolbars.py

```diff
@@ -858,6 +866,62 @@ def OnSelectMap(self, event):
 
         event.Skip()
 
+    def _topoZeroLengthCheck(self, map_name):
+        """Topology check for lines or boundaries of zero length
+
+        :param str map_name: full vector map name
+
+        :return None/tuple:
+
+        None: topology isn't available for vector map
+
+        tuple: (number of zero lines, number of zero boundaries)
+        """
+
+        map_name, mapset = ParseMapStr(map_name)
+
+        map = pointer(Map_info())
+        ret = Vect_open_old(map,
+                            c_char_p(grass.encode(map_name)),
+                            c_char_p(grass.encode(mapset)))
+        if ret == 1:
+            Vect_close(map)
+        if ret != 2:
+            return None
+
+        points = Vect_new_line_struct()
+        cats = Vect_new_cats_struct()
+        n_zero_lines = n_zero_boundaries = 0
+        nlines = Vect_get_num_lines(map)
+        for line in range(1, nlines + 1):
+            if not Vect_line_alive(map, line):
+                continue
+            ltype = Vect_get_line_type(map, line)
+            if ltype <= GV_LINES:
+                Vect_read_line(map, points, cats, line)
+                length = Vect_line_length(points)
+                if length == 0:
+                    if ltype == GV_LINES:
+                        n_zero_lines += 1
+                    elif ltype == GV_BOUNDARY:
+                        n_zero_boundaries +=1
+        Vect_close(map)
+        return n_zero_lines, n_zero_boundaries
+
+    def _cleanZeroLength(self, map_name):
+        """Clean lines or boundaries of zero length
```

Comment:
This is kind of border line case here. This calls modules, so it is okay in GUI in general and there is already v.build call here, but on the other hand, this file is toolbars.py. Something to consider. Maybe refactoring is needed if there is more stuff needed than just one v.build call.